Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Utilities.ps1

```diff
@@ -544,11 +544,108 @@ function New-ServiceFabricApplicationAction
+        if ($upgradeStatus -and ($upgradeStatus.UpgradeState -ne "RollingBackCompleted" -and $upgradeStatus.UpgradeState -ne "RollingForwardCompleted"))
+        {
+            return $false
+        }
+
+        return $true
+    }
+
+    try
+    {
+        Invoke-ActionWithDefaultRetries -Action $startAction `
+            -RetryMessage (Get-VstsLocString -Key SFSDK_RetryingUpgradeApplication) `
+            -ExceptionRetryEvaluator $exceptionRetryEvaluator `
+            -RetryableExceptions @("System.Fabric.FabricTransientException", "System.TimeoutException")
+    }
+    catch
+    {
+        # print application health status if starting upgrade did not succeed
+        Trace-ServiceFabricApplicationHealth -ApplicationName $($UpgradeParameters["ApplicationName"])
+        throw
+    }
+}
+
+function Test-ServiceFabricClusterConnectionAction
+{
+    try
+    {
+        $global:operationId = $SF_Operations.TestClusterConnection
+        $testAction = { [void](Test-ServiceFabricClusterConnection) }
+        Invoke-ActionWithDefaultRetries -Action $testAction `
+            -RetryMessage (Get-VstsLocString -Key SFSDK_RetryingTestClusterConnection) `
+            -RetryableExceptions @("System.Fabric.FabricTransientException", "System.TimeoutException")
+    }
+    catch
+    {
+        Write-Warning (Get-VstsLocString -Key SFSDK_UnableToVerifyClusterConnection)
         throw
     }
 }
 
+function Test-ServiceFabricApplicationPackageAction
+{
+    Param (
+        [string]
+        $AppPkgPath,
+
+        [string]
+        $ImageStoreConnectionString
+    )
+
+    $global:operationId = $SF_Operations.TestApplicationPackage
+    $testAction = { Test-ServiceFabricApplicationPackage -ApplicationPackagePath $AppPkgPath -ImageStoreConnectionString $ImageStoreConnectionString }
+    return Invoke-ActionWithDefaultRetries -Action $testAction `
+        -RetryMessage (Get-VstsLocString -Key SFSDK_RetryingTestAppPackage) `
+        -RetryableExceptions @("System.Fabric.FabricTransientException", "System.TimeoutException")
+}
+
+function Get-ServiceFabricClusterManifestAction
+{
+    $global:operationId = $SF_Operations.GetClusterManifest
+    $manifestAction = { Get-ServiceFabricClusterManifest }
+    return Invoke-ActionWithDefaultRetries -Action $manifestAction `
+        -RetryMessage (Get-VstsLocString -Key SFSDK_RetryingGetClusterManifest) `
+        -RetryableExceptions @("System.Fabric.FabricTransientException", "System.TimeoutException")
+}
+
+function Remove-ServiceFabricApplicationPackageAction
+{
+    Param (
+        [string]
+        $ApplicationPackagePathInImageStore,
+
+        [string]
+        $ImageStoreConnectionString
+    )
+
+    $global:operationId = $SF_Operations.RemoveApplicationPackage
+    $removeAction = { Remove-ServiceFabricApplicationPackage -ApplicationPackagePathInImageStore $applicationPackagePathInImageStore -ImageStoreConnectionString $ImageStoreConnectionString }
```

Comment:
the casing in $application starts with lower case

Reply:
#fixed